app/src/main/java/id_iot/demo/MainActivity.java

```diff
@@ -1,13 +1,66 @@
 package id_iot.demo;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.design.widget.TabLayout;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+
+import java.util.ArrayList;
 
 public class MainActivity extends AppCompatActivity {
 
+    private RecipeList recipeList;
+    private Recipe chosenRecipe;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        recipeList = new RecipeList();
+        //chosenRecipe = recipeList.getRecipes().get(0);
+
+        TabLayout tabLayout = (TabLayout) findViewById(R.id.tab_layout);
+        setUpTabs(tabLayout);
+
+        final ViewPager viewPager = (ViewPager) findViewById(R.id.screen);
+        PagerAdapter pagerAdapter = new PagerAdapter(getSupportFragmentManager(), tabLayout.getTabCount());
+        viewPager.setAdapter(pagerAdapter);
+        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
+        tabLayout.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
```

Comment:
remove this from this method as this method become too long. try to keep methods 7-10 lines long at most